### PR TITLE
OEPCIS-665: Return ProblemResponseBody when exception occurs, instead of plaintext message

### PR DESCRIPTION
**Pull request by imrohanshah**
- Add ObjectMapper in VersionTransformer to generate ProblemResponseBody's JSON representation to return.

### CODE REVIEW

**Review by sboeckelmann — CHANGES_REQUESTED** [review]
I am not convinced that this is the right place to take care of the original issue.
Code should throw Exceptions instead of swallowing them, otherwise it's not possible to setup http error status code.
Exception handling should be done in HTTP Service providers:
e.g.
openepcis-format-converter-service/converter-service-rest/src/main/java/io/openepcis/epc/converter/resource/DocumentConverterResource.java

File: src/main/java/io/openepcis/convert/exception/FormatConverterException.java [hunk 1]
```diff
@@ -15,6 +15,8 @@
  */
 package io.openepcis.convert.exception;
 
+import io.openepcis.model.rest.ProblemResponseBody;
+
```
[the remark this answers is not in the document]
Comment (imrohanshah) [reply src/main/java/io/openepcis/convert/exception/FormatConverterException.java:19]: This import is not required anymore. We will remove it asap.

File: src/main/java/io/openepcis/convert/VersionTransformer.java [hunk 2]
```diff
@@ -393,13 +394,10 @@ private InputStream fromXmlToXml(final InputStream inputDocument) {
                             xmlOutputStream.close();
                         } catch (Exception e) {
                             try {
-                                xmlOutputStream.write(e.getMessage().getBytes());
+                                xmlOutputStream.write(objectMapper.writeValueAsBytes(ProblemResponseBody.fromException(e)));
                                 xmlOutputStream.close();
```
[the remark this answers is not in the document]
Comment (imrohanshah) [reply src/main/java/io/openepcis/convert/VersionTransformer.java:398]: This was a missout, we will fix it!

File: src/main/java/io/openepcis/convert/VersionTransformer.java [hunk 3]
```diff
@@ -393,13 +394,10 @@ private InputStream fromXmlToXml(final InputStream inputDocument) {
...
-                            } finally {
-                                throw new FormatConverterException(
-                                        "Exception occurred during the conversion of XML document to XML document  : "
-                                                + e.getMessage(),
-                                        e);
+                            } catch (IOException ioe) {
+                                log.warn("Couldn't write or close the stream", ioe);
```
[the remark this answers is not in the document]
Comment (imrohanshah) [reply src/main/java/io/openepcis/convert/VersionTransformer.java:400]: This exception was thrown in ExecutorService. The method returns a PipedInputStream, to which the executorService writes. If there is an exception, the caller code still receives the stream and the Exception is not getting Propagated from what we have noticed. So we decided that we could remove that throw statement.

And for this reason, we have removed thrown exception from other similar places where they were being thrown in ExecutorService and did not get propagated to the ExceptionMapper.

[the remark this answers is not in the document]
Comment (sboeckelmann) [reply src/main/java/io/openepcis/convert/VersionTransformer.java:400]: Yes, okay. Refactoring it to return Multi<byte[]> would be too much effort.

**Pull request merged**